File: packages/shared/types/pfsense.ts
export type RuleAction = 'block' | 'pass' | 'reject'
export type RuleProtocol = 'tcp' | 'udp' | 'tcp/udp' | 'icmp' | 'any'

export interface PfSenseRule {
  rule_id: string
  action: RuleAction
  protocol: RuleProtocol
  src_ip: string
  dst_ip: string | null
  dst_port: number | null
  description: string
  created_at: string
  created_by: string   // analysis_id or 'manual'
  active: boolean
}

export interface BlockRequest {
  ip: string
  reason: string
  analysis_id: string | null
  duration_hours: number | null  // null = permanent
}

export interface BlockResponse {
  success: boolean
  rule_id: string | null
  message: string
}

File: backend/models/pfsense.py
from pydantic import BaseModel, ConfigDict


class PfSenseRule(BaseModel):
    model_config = ConfigDict(extra='ignore')

    rule_id: str
    action: str
    protocol: str
    src_ip: str
    dst_ip: str | None
    dst_port: int | None
    description: str
    created_at: str
    created_by: str
    active: bool


class BlockRequest(BaseModel):
    model_config = ConfigDict(extra='ignore')

    ip: str
    reason: str
    analysis_id: str | None
    duration_hours: int | None


class BlockResponse(BaseModel):
    model_config = ConfigDict(extra='ignore')

    success: bool
    rule_id: str | None
    message: str
